### PR TITLE
Feat: added cart mamagement features , add to cart , remove fromm , i…

### PR DESCRIPTION
**Pull request by Divya70**
Added Cart Mananegment Featue 

**Add To Cart**
**Delete Cart **
**Increase / Decrease Quantity**
**Price Details**
**added .env file**
**remove .env file from gitignore**

### CODE REVIEW

File: src/components/Cart/PriceDetails/PriceDetails.js [hunk 1]
```diff
@@ -1,30 +1,48 @@
...
+  const getPriceBill = (cartItem) => {
+    const currentQty = cartItem.reduce(
+      (acc, curr) => acc + Number(curr.qty),
+      0
+    );
+    const currentPrice = cartItem.reduce(
+      (acc, curr) => (acc += Number(curr.price) * Number(curr.qty)),
+      0
+    );
+    const totalSaving = currentPrice + 10;
```
Comment (Ayushi20-19) [inline src/components/Cart/PriceDetails/PriceDetails.js:15]: how come saving will be current Price + 10?

Comment (Divya70) [reply src/components/Cart/PriceDetails/PriceDetails.js:15]: okay , so i am giving 50 rs discount in hard code and 40 rs. delevery charge . so i am substracting 50-40=10

File: src/components/Cart/ProductDetails/ProductDetails.js [hunk 2]
```diff
@@ -1,34 +1,100 @@
...
+
+  const token = localStorage.getItem("token");
+  const removeCartHandler = async () => {
+    try {
+      const removeCartRes = await axios.delete(`/api/user/cart/${item._id}`, {
+        headers: { authorization: token },
+      });
+      dispatch({ type: "REMOVE_FROM_CART", payload: removeCartRes.data.cart });
+    } catch (err) {
+      console.log(err.response);
```
Comment (RahullRawat) [inline src/components/Cart/ProductDetails/ProductDetails.js:17]: You can remove these `console.log`

File: src/components/Cart/ProductDetails/ProductDetails.js [hunk 3]
```diff
@@ -1,34 +1,100 @@
...
+          },
+        },
+        {
+          headers: { authorization: token },
+        }
+      );
+
+      dispatch({ type: "INCREASE_QUANTITY", payload: quantityRes.data.cart });
+    } catch (error) {
+      console.log("Error:", error);
```
Comment (RahullRawat) [inline src/components/Cart/ProductDetails/ProductDetails.js:36]: You can remove these `console.log`

File: src/components/Cart/ProductDetails/ProductDetails.js [hunk 4]
```diff
@@ -1,34 +1,100 @@
...
+          },
+        },
+        {
+          headers: { authorization: token },
+        }
+      );
+
+      dispatch({ type: "DECREASE_QUANTITY", payload: quantityRes.data.cart });
+    } catch (error) {
+      console.log("Error:", error);
```
Comment (RahullRawat) [inline src/components/Cart/ProductDetails/ProductDetails.js:55]: You can remove these `console.log`

File: src/components/Products/Products.js [hunk 5]
```diff
@@ -1,18 +1,34 @@
...
+        headers: { authorization: token },
+      }
+    );
+    if (addToCartResponse.status === 201) {
+      dispatch({
+        type: "ADD_TO_CART",
+        payload: addToCartResponse.data.cart,
+      });
+    } else {
+      console.log("error");
```
Comment (RahullRawat) [inline src/components/Products/Products.js:24]: You can remove these `console.log`

File: src/pages/Cart/Cart.js [hunk 6]
```diff
@@ -1,15 +1,49 @@
...
+  const { state, dispatch } = useProduct();
+  const getProductInCart = async () => {
+    try {
+      const res = await axios.get("/api/user/cart", {
+        headers: { authorization: localStorage.getItem("token") },
+      });
+      if (res.status === 200) {
+        dispatch({ type: "GET_CART", payload: res.data.cart });
+      } else {
+        console.log("err:", res);
```
Comment (RahullRawat) [inline src/pages/Cart/Cart.js:16]: You can remove these `console.log`

File: src/pages/Cart/Cart.js [hunk 7]
```diff
@@ -1,15 +1,49 @@
...
+      }
+    } catch (error) {
+      console.log("Error:", error);
```
Comment (RahullRawat) [inline src/pages/Cart/Cart.js:19]: You can remove these `console.log`

File: src/components/Cart/PriceDetails/PriceDetails.js (hunk 1 above)
Comment (Ayushi20-19) [reply src/components/Cart/PriceDetails/PriceDetails.js:15]: The saving amount will be the amount that the user saves on a particular product? or you are trying to show something else?

File: src/pages/Login/Login.js [hunk 8]
```diff
@@ -29,18 +29,20 @@ const Login = () => {
...
+            token: res.data.encodedToken,
+            user: res.data.foundUser,
+          },
         });
       } else {
         alert("Please Enter Correct Email and Password!");
       }
     } catch (error) {
-      console.log("Erro:", error);
+      console.log("Error:", error.res);
```
Comment (RahullRawat) [inline src/pages/Login/Login.js:45]: You can remove these `console.log`

**Review by RahullRawat — APPROVED** [review]
Your cart functionality is working fine. 

**These are some small suggestions:-**
- Your navbar  is displaying `logout` button, even if i am not logged in. Please check this
- If user is not loggrd in, then user sholud not be able to go to cart page.
- When cart is empty, you can display some msg like `Your cart is empty`.
- Also you can give `cursor : pointer` to your buttons

File: src/pages/Cart/Cart.js (hunk 7 above)
Comment (Divya70) [reply src/pages/Cart/Cart.js:19]: all console.log are for catching error

**Review by Tejinder15 — APPROVED** [review]

File: src/components/Cart/PriceDetails/PriceDetails.js (hunk 1 above)
Comment (Divya70) [reply src/components/Cart/PriceDetails/PriceDetails.js:15]: yes , i am trying to show the discount amount which user has saved on perticular product

**Pull request merged**